flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/DefaultSlotPoolServiceSchedulerFactory.java

```diff
@@ -232,7 +232,9 @@ private static JobManagerOptions.SchedulerType getSchedulerType(
                     .getOptional(JobManagerOptions.SCHEDULER)
                     .orElse(
                             System.getProperties()
-                                            .containsKey("flink.tests.enable-adaptive-scheduler")
+                                                    .containsKey(
+                                                            "flink.tests.enable-adaptive-scheduler")
+                                            && jobType == JobType.STREAMING
```

Comment:
I would propose to update the fallback logics in `fromConfiguration()` too. "Falling back to DefaultScheduler if it is a batch job" is no longer expected after we have changed AdaptiveBatchScheduler to be the default scheduler for batch jobs.